### PR TITLE
refactor: unify cpu gpu into device

### PR DESCRIPTION
**Pull request by bwanglzu**
Resolve inconsistency between `fit` end `get_model`, in `fit` we use `cpu`, in `get_model` we use `gpu`. Better to align with Docarray `device='cpu'` or `device='cuda'`.

This will not be a breaking change for internal users. 

This will be a breaking change for ppl install `finetuner-full`  when calling `get_model`.

A follow up ticket has been created to remove `cpu` in `0.7.0` #578 

```diff
# training time
run = finetuner.fit(
    ...,
-   cpu=False,  # cpu is still supported, no breaking change, we sunset `cpu` after 0.7
+   device='cuda', # note, now `cuda` is the default device.
)

# inference time
inference_engine = finetuner.get_model(
     ...,
-    gpu=True, 
+    device='cuda',  # note, `cpu` is the default device unless user specify `cuda`.
)
```

---

- [ ] This PR references an open issue
- [x] I have added a line about this change to CHANGELOG

### CODE REVIEW

File: finetuner/experiment.py [hunk 1]
```diff
@@ -165,14 +167,24 @@ def create_run(
             **kwargs,
         )
 
-        cpu = kwargs.get(CPU, True)
+        device = kwargs.get(DEVICE, 'cuda')
+        if device == 'cpu' and not kwargs.get(CPU, True):
```
Comment (gmastrapas) [inline finetuner/experiment.py:171]: can you explain the logic here?

Comment (gmastrapas) [reply finetuner/experiment.py:171]: so it should be `if device == 'cuda' and kwargs.get(CPU, True):`?

Comment (bwanglzu) [reply finetuner/experiment.py:171]: yes it's more clear

**Review by gmastrapas — CHANGES_REQUESTED** [review]

File: finetuner/__init__.py [hunk 2]
```diff
@@ -338,7 +347,7 @@ def get_model(
         TorchInferenceEngine,
     )
 
-    if gpu:
+    if device == 'cuda' and is_onnx:
         warnings.warn(
             message='You are using cuda device for ONNX inference, please consider'
             'call `pip install onnxruntime-gpu` to speed up inference.',
```
Comment (gmastrapas) [inline finetuner/__init__.py:353]: consider calling

File: finetuner/__init__.py [hunk 3]
```diff
@@ -167,10 +168,16 @@ def fit(
     :param output_dim: The expected output dimension as `int`.
         If set, will attach a projection head.
     :param cpu: Whether to use the CPU. If set to `False` a GPU will be used.
```
Comment (gmastrapas) [inline finetuner/__init__.py:170]: can you add here that this arg will be deprecated, and you should use `device` instead?

**Review by gmastrapas — CHANGES_REQUESTED** [review]

**Review by LMMilliken — APPROVED** [review]

**Review by gmastrapas — APPROVED** [review]

**Pull request merged**